common/Geometry2d/Pose.hpp

```diff
@@ -0,0 +1,272 @@
+#pragma once
+
+#include "Point.hpp"
+#include "TransformMatrix.hpp"
+
+namespace Geometry2d {
+
+/**
+ * Represents a pose in 2d space: (x, y, theta). Uses double-precision floating
+ * point numbers for the coordinates, and can be used to represent the position
+ * and orientation of a robot in the plane.
+ */
+class Pose {
+public:
+    /**
+     * Default constructor - zero-initialize
+     */
+    Pose() { _pose = Eigen::Vector3d::Zero(); }
+
+    /**
+     * Point-heading constructor
+     */
+    Pose(Point position, double heading)
+        : _pose(position.x(), position.y(), heading) {}
+
+    /**
+     * Component-wise constructor
+     */
+    Pose(double x, double y, double h) : _pose(x, y, h) {}
+
+    /**
+     * Implicit conversion from Eigen::Vector3d
+     */
+    Pose(const Eigen::Vector3d& other) : _pose(other) {}
+
+    /**
+     * Copy-constructor - default
+     */
+    Pose(const Pose& other) = default;
+
+    /**
+     * Compute the pose specified using this pose as coordinates in a frame of
+     * reference specified by `other`, in the global space.
+     *
+     * i.e.
+     *
+     * x:
+     *       |  x
+     *       | /
+     *       |/
+     * ------|-------
+     *       |
+     *       |
+     *       |
+     *
+     * y:
+     *     |
+     *     |
+     * ----|----
+     *     |\
+     *     | y
+     *
+     * y.withOrigin(x):
+     *       |  x
+     *       | / \
+     *       |/   y
+     * ------|------
+     *       |
+     */
+    Pose withOrigin(Pose other) const {
+        double sh = std::sin(other.heading()), ch = std::cos(other.heading());
```

Comment:
Aren't sh and ch not used?

Reply:
Uh, oops...I originally implemented it using Eigen and then realized we already had rotation baked into geometry so I deleted it. My bad, I'll fix.